cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,5263 @@
+    "\n",
+    "We can now use Cleanlab's [Trustworthy Language Model](https://cleanlab.ai/blog/trustworthy-language-model/) to tag our document chunks with the correct document topic that we can later use to enhance our retrieval process of fetching the correct context from our vector DB."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a4cc3362",
+   "metadata": {},
+   "source": [
+    "Cleanlab's Trustworthy Language Model (TLM) is a more reliable LLM that gives high-quality outputs and indicates when it is unsure of the answer to a question, making it suitable for applications where unchecked hallucinations are a show-stopper."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e7e0a4b0",
+   "metadata": {},
+   "source": [
+    "### Installing Cleanlab TLM\n",
+    "\n",
+    "Using TLM requires a [Cleanlab](https://app.cleanlab.ai/) account. Sign up for one [here](https://cleanlab.ai/signup/) if you haven't yet. If you've already signed up, check your email for a personal login link.\n",
+    "\n",
+    "Cleanlab's Python client can be installed using pip below (but the correct version used in this tutorial was installed in `requirements.txt` at the beginning of this notebook):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "467d6071",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install -U cleanlab-studio"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e5308022",
+   "metadata": {},
+   "source": [
+    "## Using the TLM\n",
+    "\n",
+    "You can use the TLM pretty much like any other LLM API:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "0f764f6d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from cleanlab_studio import Studio\n",
+    "\n",
+    "key = \"Please enter your Cleanlab API key\"\n",
+    "\n",
+    "# Get API key from here: https://app.cleanlab.ai/account after creating an account.\n",
+    "studio = Studio(key)\n",
+    "\n",
+    "# Use the most powerful OpenAI model with TLM\n",
+    "tlm = studio.TLM(quality_preset=\"high\", options={\"model\": \"gpt-4o\"})\n",
+    "\n",
+    "output = tlm.prompt(\"<your prompt>\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "47c191e8",
+   "metadata": {},
+   "source": [
+    "Let's now use Cleanlab's TLM to do zero-shot classification and classify the text (tag) into different topics. We will make use of code from [Cleanlab's TLM Zero-Shot Classification Tutorial](https://help.cleanlab.ai/tutorials/zero_shot_classification/) to do this. This includes the two helper functions `parse_category()` and `classify()` that can be found below."
```

Comment:
you dont need to write classification all of the time, you can just write classification.